modules/geoedgeRtdProvider.js

```diff
@@ -0,0 +1,213 @@
+  if (wrapperReady) {
+    return success(wrapper);
+  }
+  ajax(WRAPPER_URL, success);
+}
+
+/**
+ * sets the wrapper and calls preload client
+ * @param {string} responseText
+ */
+export function setWrapper(responseText) {
+  wrapperReady = true;
+  wrapper = responseText;
+}
+
+/**
+ * preloads the client
+  * @param {string} key
+ */
+export function preloadClient(key) {
+  let link = document.createElement('link');
+  link.rel = 'preload';
+  link.as = 'script';
+  link.href = getClientUrl(key);
+  link.onload = () => { preloaded = true };
+  insertElement(link);
+}
+
+/**
+ * creates identity function for string replace without special replacement patterns
+ * @param {string} str
+ * @return {function}
+ */
+function replacer(str) {
+  return function () {
+    return str;
+  }
+}
+
+export function wrapHtml(wrapper, html) {
+  return wrapper.replace(HTML_PLACEHOLDER, replacer(html));
+}
+
+/**
+ * generate macros dictionary from bid response
+ * @param {Object} bid
+ * @param {string} key
+ * @return {Object}
+ */
+function getMacros(bid, key) {
+  return {
+    '${key}': key,
+    '%%ADUNIT%%': bid.adUnitCode,
+    '%%WIDTH%%': bid.width,
+    '%%HEIGHT%%': bid.height,
+    '%%PATTERN:hb_adid%%': bid.adId,
+    '%%PATTERN:hb_bidder%%': bid.bidderCode,
+    '%_isHb!': true,
+    '%_hbcid!': bid.creativeId || '',
+    '%%PATTERN:hb_pb%%': bid.pbHg,
+    '%%SITE%%': location.hostname,
+    '%_pimp%': PV_ID
+  };
+}
+
+/**
+ * replace macro placeholders in a string with values from a dictionary
+ * @param {string} wrapper
+ * @param {Object} macros
+ * @return {string}
+ */
+function replaceMacros(wrapper, macros) {
+  let key;
+  for (key in macros) {
+    wrapper = wrapper.replace(key, replacer(macros[key]));
+  }
+  return wrapper;
+}
+
+/**
+ * build final creative html with creative wrapper
+ * @param {Object} bid
+ * @param {string} wrapper
+ * @param {string} html
+ * @return {string}
+ */
+function buildHtml(bid, wrapper, html, key) {
+  let macros = getMacros(bid, key);
+  wrapper = replaceMacros(wrapper, macros);
+  return wrapHtml(wrapper, html);
+}
+
+/**
+ * muatates the bid ad property
+ * @param {Object} bid
+ * @param {string} ad
+ */
+function mutateBid(bid, ad) {
+  bid.ad = ad;
+}
+
+/**
+ * wraps a bid object with the creative wrapper
+ * @param {Object} bid
+ * @param {string} key
+ */
+export function wrapBidResponse(bid, key) {
+  let wrapped = buildHtml(bid, wrapper, bid.ad, key);
+  mutateBid(bid, wrapped);
+}
+
+/**
+ * checks if bidder's bids should be monitored
+ * @param {string} bidder
+ * @return {boolean}
+ */
+function isSupportedBidder(bidder, paramsBidders) {
+  return isEmpty(paramsBidders) || paramsBidders[bidder] === true;
+}
+
+/**
+ * checks if bid should be monitored
+ * @param {Object} bid
+ * @return {boolean}
+ */
+function shouldWrap(bid, params) {
+  let supportedBidder = isSupportedBidder(bid.bidderCode, params.bidders);
+  let wap = params.wap ? preloaded : true;
```

Comment:
The `wap` parameter is defaulting to `true`. But in your `md` file, you've mentioned the default value is `false`. May wanna look this through...

Reply:
Renamed to `donePreload` for clarity, the parameter value determines if the preload state should be taken into consideration in the return value of this function.